client/src/Form.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+class Form extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: '' };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  handleSubmit(event) {
+    alert('A name was submitted: ' + this.state.value);
+    event.preventDefault();
+  }
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.handleSubmit}>
+          <label>
+            Name:
+          <input type="text" value={this.state.value} onChange={this.handleChange} />
+          </label>
+          <label>
+            number:
+          <input type="text" value={this.state.value} onChange={this.handleChange} />
+          </label>
+          <label>
+            Email:
+          <input type="text" value={this.state.value} onChange={this.handleChange} />
+          </label>
+          <label>
+            Message:
+          <input type="text" value={this.state.value} onChange={this.handleChange} />
+          </label>
+          <input type="submit" value="Contact Agent" />
```

Comment:
@SduongPlanD  You need a onClick event here.
            <button type="submit" onClick={handlePaymentInfoClick}  ">Contact Agent</button>

Hers an example of a handleSubmit method

    handleCompleteClick (e) {
        e.preventDefault();
        // creat an object to store the information that will be posted to... somewhere
        var data = {
            name: this.state.information.name,
            number: this.state.information.number,
            email: this.state.information.email, 
            message: this.state.information.message, 
        }
        
       // post customer information, in this example, this would be an ajax request
        postCustomerData(data, (err) => {
            if (err) {
                console.log(err)
                return
            }
        })
        // set state to... somthing 
        this.setState( {
            // do something 
        })
    }

Reply:
updated